File: app/src/main/java/com.example.safehome/Camera/Camera.java
package Camera;

public class Camera {

	/**
	 * camera id
	 */
	private string id = null;
	private string type = null;
	/**
	 * camera location
	 */
	private string location = null;
	private string fieldOfView = null;
	private boolean twoWayAudio = null;
	private boolean nightVision = null;
	private string resolution = null;
	/**
	 * Defaults to false.
	 */
	private boolean hasPushNotifications = false;
	private string connectivity = null;
	private ConcreteCamera state;
	/**
	 * Defines the storage object.
	 */
	private Storage storage = null;

	public void setType(string type) {
		this.type = type;
	}

	public void setState(ConcreteCamera state) {
		this.state = state;
	}

	public ConcreteCamera saveState() {
		// TODO - implement Camera.saveState
		throw new UnsupportedOperationException();
	}

	/**
	 * 
	 * @param cameraMomemto
	 */
	public void restoreState(CameraMomemto cameraMomemto) {
		// TODO - implement Camera.restoreState
		throw new UnsupportedOperationException();
	}

}

File: app/src/main/kotlin/com.example.safehome/Camera/Camera.kt
package com.example.safehome.Camera

import org.slf4j.LoggerFactory

class Camera {
    /**
     * camera id
     */
    private val id: String? = null
    private var type: String? = null

    /**
     * camera location
     */
    private val location: String? = null
    private val fieldOfView: String? = null
    private val twoWayAudio = false
    private val nightVision = false
    private val resolution: String? = null

    /**
     * Defaults to false.
     */
    private val hasPushNotifications = false
    private val connectivity: String? = null
    private var state: Camera? = null

    /**
     * Defines the storage object.
     */
    private val storage: Storage<*>? = null

    /**
     * Defines the camera type.
     * @param type the camera type.
     */
    fun setType(type: String?) {
        this.type = type
    }

    /**
     * Saves a new state of the Object.
     * @param newState the Object state to save
     */
    fun setState(newState: Camera?) {
        state = newState
    }

    /**
     * Retrieve the save state of the Object
     * @return the saved state of the Object
     */
    fun saveState(): Camera? {
        return state
    }

    /**
     * Create a restore state from the CameraMomento.
     * @param cameraMomemto the saved state of an Object.
     */
    fun restoreState(cameraMomemto: CameraMomento?) {
        // TODO - implement Camera.restoreState
//		throw new UnsupportedOperationException();
    }

    companion object {
        private val log = LoggerFactory.getLogger(Camera::class.java)
    }
}